src/test/compile-fail/cast-rfc0401.rs

```diff
@@ -59,10 +59,12 @@ fn main()
     //~^ ERROR casting
     //~^^ HELP through a usize first
     let _ = 3_i32 as bool;
-    //~^ ERROR cannot cast as `bool`
+    //~^ ERROR cannot cast as `bool` [E0054]
```

Comment:
@jonathandturner 
In face, if I changed

    //~^ ERROR cannot cast as `bool` [E0054]

to

    //~^ ERROR E0054 cannot cast as `bool`

I got the error message like this:
```
unexpected errors (from JSON output): [
    Error {
        line_num: 61,
        kind: Some(
            Error
        ),
        msg: "61:13: 61:26: cannot cast as `bool` [E0054]"
    }
]

not found errors (from test file): [
    Error {
        line_num: 61,
        kind: Some(
            Error
        ),
        msg: "E0054 cannot cast as `bool`"
    }
]
```

I believe it makes sense to leave `[E0054]` on the right.